src/Cardano/HttpBridge/Api.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | An API specification for the Cardano HTTP Bridge.
+module Cardano.HttpBridge.Api
+    ( Api
+    , api
+    ) where
+
+import Cardano.Network.Name
+    ( NetworkName )
+import Cardano.Wallet.Primitive
+    ( Block, BlockHeader, EpochId, Hash )
+import Crypto.Hash.Algorithms
+    ( Blake2b_256 )
+import Servant
+    ( Proxy (..) )
+import Servant.API
+    ( (:<|>), (:>), Capture, Get )
+import Servant.Extra.ContentTypes
+    ( CBOR, Packed, WithHash )
+
+api :: Proxy Api
+api = Proxy
+
+type Api
+    =    GetBlockByHash
+    :<|> GetEpochById
+    :<|> GetTipBlockHeader
+
+-- | Retrieve a block identified by the unique hash of its header.
+type GetBlockByHash
+    =  Capture "networkName" NetworkName
+    :> "block"
+    :> Capture "blockHeaderHash" (Hash "BlockHeader")
+    :> Get '[CBOR] Block
+
+-- | Retrieve all the blocks for the epoch identified by the given integer ID.
+type GetEpochById
+    =  Capture "networkName" NetworkName
+    :> "epoch"
+    :> Capture "epochId" EpochId
+    :> Get '[Packed CBOR] [Block]
+
+-- | Retrieve the header of the latest known block.
+type GetTipBlockHeader
+    = Capture "networkName" NetworkName
+    :> "tip"
+    :> Get '[WithHash Blake2b_256 CBOR] (Hash "BlockHeader", BlockHeader)
```

Comment:
Okay. That is nice :) 

One remark though, I'd rather use newtypes around those various primitive types here to avoid orphan instances. This is going to byte us later on.

Reply:
Fair enough! I've added newtypes and removed all the orphan instances.